tests/Dialect/Openfhe/Transforms/configure_crypto_context_complex.mlir

```diff
@@ -1,46 +1,20 @@
-// RUN: heir-opt --openfhe-configure-crypto-context=entry-function=complex_func %s | FileCheck %s
+// RUN: heir-opt --secretize --wrap-generic --cse --canonicalize --secret-insert-mgmt-bgv --secret-distribute-generic --secret-to-bgv=poly-mod-degree=32 --bgv-to-openfhe --openfhe-configure-crypto-context=entry-function=complex_func %s | FileCheck %s
 
-!Z1095233372161_i64_ = !mod_arith.int<1095233372161 : i64>
-!Z65537_i64_ = !mod_arith.int<65537 : i64>
+!ty = tensor<32xi16>
 
-!rns_L0_ = !rns.rns<!Z1095233372161_i64_>
-
-#ring_Z65537_i64_1_x32_ = #polynomial.ring<coefficientType = !Z65537_i64_, polynomialModulus = <1 + x**32>>
-#ring_rns_L0_1_x32_ = #polynomial.ring<coefficientType = !rns_L0_, polynomialModulus = <1 + x**32>>
-
-#full_crt_packing_encoding = #lwe.full_crt_packing_encoding<scaling_factor = 0>
-#key = #lwe.key<>
-
-#modulus_chain_L5_C0_ = #lwe.modulus_chain<elements = <1095233372161 : i64, 1032955396097 : i64, 1005037682689 : i64, 998595133441 : i64, 972824936449 : i64, 959939837953 : i64>, current = 0>
-
-#plaintext_space = #lwe.plaintext_space<ring = #ring_Z65537_i64_1_x32_, encoding = #full_crt_packing_encoding>
-
-#ciphertext_space_L0_ = #lwe.ciphertext_space<ring = #ring_rns_L0_1_x32_, encryption_type = lsb>
-
-!plain_ty = !lwe.new_lwe_plaintext<application_data = <message_type = tensor<32xi16>>, plaintext_space = #plaintext_space>
-!ctxt_ty = !openfhe.crypto_context
-!in_ty = !lwe.new_lwe_ciphertext<application_data = <message_type = tensor<32xi16>>, plaintext_space = #plaintext_space, ciphertext_space = #ciphertext_space_L0_, key = #key, modulus_chain = #modulus_chain_L5_C0_>
-!out_ty = !lwe.new_lwe_ciphertext<application_data = <message_type = i16>, plaintext_space = #plaintext_space, ciphertext_space = #ciphertext_space_L0_, key = #key, modulus_chain = #modulus_chain_L5_C0_>
-
-func.func @complex_func(%arg0: !ctxt_ty, %arg1: !in_ty, %arg2: !in_ty, %cond: i1) -> !out_ty {
+func.func @complex_func(%arg0 : !ty {secret.secret}, %arg1 : !ty {secret.secret}) -> !ty {
   %cst = arith.constant dense<[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31]> : tensor<32xi16>
-  %plain = lwe.rlwe_encode %cst {encoding = #full_crt_packing_encoding, ring = #ring_Z65537_i64_1_x32_} : tensor<32xi16> -> !plain_ty
+  %0 = arith.muli %arg0, %arg1 : !ty
+  %1 = arith.addi %0, %arg0 : !ty
+  %2 = arith.muli %0, %cst : !ty
+
+  %4 = arith.muli %arg0, %cst : !ty
+  %5 = arith.muli %4, %arg1 : !ty
+  %6 = arith.subi %5, %4 : !ty
+  %7 = arith.muli %5, %cst : !ty
 
-  %ret = scf.if %cond -> !out_ty {
```

Comment:
I know why we can't lower an oblivious `scf.if` yet (#825, another context-aware-type-conversion issue) but can you remind me what breaks for plaintext `scf.if`?

Reply:
plaintext `scf.if` persists, so it is still a big op and `secret-distribute-generic` can not divide it, hence the `secret-to-<scheme>` could not recognize such pattern.

```mlir
func.func @add(%arg0: tensor<4xi16> {secret.secret}, %arg1: tensor<4xi16> {secret.secret}, %cond : i1) -> tensor<4xi16> {
  %0 = scf.if %cond -> tensor<4xi16> {
    %1 = arith.muli %arg0, %arg1 : tensor<4xi16>
    scf.yield %1 : tensor<4xi16>
  } else {
    %2 = arith.addi %arg0, %arg1 : tensor<4xi16>
    scf.yield %2 : tensor<4xi16>
  }
  return %0 : tensor<4xi16>
}
```

You could try this with `--mlir-to-secret-arithmetic --secret-insert-mgmt-bgv --secret-distribute-generic --secret-to-bgv=poly-mod-degree=4`, and will see a `failed to legalize operation 'secret.generic'` containing `scf.if`